### PR TITLE
Templates for enabling giftless service

### PR DESCRIPTION
**Pull request by zelima**
Giftless a Python implementation of a Git LFS Server https://github.com/datopian/giftless. 

PR includes templates to deploy the Git LFS server on ckan-cloud-clusters

- Secrets for mounting private and public RSA keys
  - Also, mount in ckan pods for https://github.com/datopian/ckanext-authz-service
- Configmap for giftless config
- deployment and service templates
- Proxy giftless requests (for browser requests)

### CODE REVIEW

File: ckan/templates/ckan-deployment.yaml [hunk 1]
```diff
@@ -130,6 +130,16 @@ spec:
         - name: ckan-conf-templates
           mountPath: /etc/ckan-conf/templates
           readOnly: true
+        {{ if .Values.enableGiftless }}
+        - mountPath: /etc/ssh/jwt-rs256.key
+          name: jwt-key
+          readOnly: true
+          subPath: jwt-rs256.key
```
Comment (akariv) [inline ckan/templates/ckan-deployment.yaml]: is the subPath needed here?

File: ckan/templates/giftless-secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{ if .Values.enableGiftless }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: jwt-keys
+data:
+  jwt-rs256.key: {{ .Values.jwtKey }}
```
Comment (akariv) [inline ckan/templates/giftless-secrets.yaml]: perhaps a more descriptive name for these keys (jwtKey is a bit vague)

**Review by akariv — APPROVED** [review]
Please also update the values.yaml documntation with the new values keys (in the CCO repo)

File: ckan/templates/ckan-deployment.yaml (hunk 1 above)
Comment (zelima) [reply ckan/templates/ckan-deployment.yaml]: Not sure, probably not, but it works as with it. Anyway, will try removing

**Comment by zelima** [comment]
@akariv https://github.com/datopian/ckan-cloud-operator/pull/130

**Pull request merged**